transport/internal/tls/muxlistener/listener.go

```diff
@@ -138,8 +138,8 @@ func (l *listener) serve() {
 	defer func() {
 		cancel()
 		wg.Wait()
-		close(l.stoppedChan)
 		close(l.connChan)
+		close(l.stoppedChan)
```

Comment:
Does this order matter?

Reply:
It doesn't change anything for determinism, it's more of a logical change since Close waits on stoppedChan before returning, hence keeping it as last instruction in serve. 
```
func (l *listener) Close() error {
	var err error
	l.closeOnce.Do(func() {
		err = l.Listener.Close()
		close(l.stopChan)
		<-l.stoppedChan
	})
	return err
}
```

Reply:
Yea this is more about the sequence, the above order makes Close() wait until the connChan channel is closed which was incorrectly skipped earlier